### PR TITLE
[bitnami/influxdb] make more flexible when using volumePermissions

### PR DESCRIPTION
**Pull request by mkerins**
I had some issues installing InfluxDB using the `high-availability` architecture but not when using PostgreSQL in a similar configuration. This was on a multi-node microk8s cluster using `microk8s.io/hostpath`. Essentially I needed more control over how the init-chmod-data container was run. I used the relevant bits from the PostgreSQL chart to achieve this.

**Description of the change**

Add volumePermissions.securityContext.runAsUser

**Benefits**

More flexibility when deploying chart on different distributions with different storage provisioners

**Possible drawbacks**

None that I'm aware of

**Checklist**
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

**Comment by rafariossaa** [comment]
This LGTM, but @juan131 Would you mind to take a look ?

**Comment by mkerins** [comment]
thanks @rafariossaa

File: bitnami/influxdb/Chart.yaml [hunk 1]
```diff
@@ -15,4 +15,4 @@ maintainers:
 name: influxdb
 sources:
   - https://github.com/bitnami/bitnami-docker-influxdb
-version: 0.4.7
+version: 0.5.7
```
Comment (juan131) [inline bitnami/influxdb/Chart.yaml:18]: This shouldn't be `0.5.7`, but `0.5.0` instead

File: bitnami/influxdb/templates/influxdb/statefulset-high-availability.yaml [hunk 2]
```diff
@@ -64,13 +64,25 @@ spec:
           image: {{ include "influxdb.volumePermissions.image" . }}
           imagePullPolicy: {{ .Values.volumePermissions.image.pullPolicy | quote }}
           command:
-            - sh
-            - -c
+            - /bin/bash
+            - -cx
```
Comment (juan131) [inline bitnami/influxdb/templates/influxdb/statefulset-high-availability.yaml:68]: Please keep the original shell

File: bitnami/influxdb/templates/influxdb/statefulset-high-availability.yaml [hunk 3]
```diff
@@ -64,13 +64,25 @@ spec:
...
             - |
+              {{- if .Values.persistence.enabled }}
```
Comment (juan131) [inline bitnami/influxdb/templates/influxdb/statefulset-high-availability.yaml:70]: This is not necessary. You're already checking this is `true` on line 61.

File: bitnami/influxdb/templates/influxdb/statefulset-high-availability.yaml [hunk 4]
```diff
@@ -64,13 +64,25 @@ spec:
...
               mkdir -p /bitnami/influxdb/{data,meta,wal}
-              chown -R "{{ .Values.influxdb.securityContext.runAsUser }}:{{ .Values.influxdb.securityContext.fsGroup }}" /bitnami/influxdb/{data,meta,wal}
+              chmod 700 /bitnami/influxdb/{data,meta,wal}
+              find /bitnami/influxdb/{data,meta,wal} -mindepth 1 -maxdepth 1 | \
+              {{- if eq ( toString ( .Values.volumePermissions.securityContext.runAsUser )) "auto" }}
+                xargs chown -R `id -u`:`id -G | cut -d " " -f2`
+              {{- else }}
+                xargs chown -R {{ .Values.influxdb.securityContext.runAsUser }}:{{ .Values.influxdb.securityContext.fsGroup }}
+              {{- end }}
```
Comment (juan131) [inline bitnami/influxdb/templates/influxdb/statefulset-high-availability.yaml:78]: Why using this `find .... | xargs chown -R ...` approach? This approach can be useful when you're excluding some directories or files, but I think that's not the case.

I mean, is there any actual difference between this and `chown -R ... /bitnami/influxdb/{data,meta,wal}`

File: bitnami/influxdb/README.md [hunk 5]
```diff
@@ -386,7 +387,7 @@ As the images run as non-root by default, it is necessary to adjust the ownershi
 By default, the chart is configured to use Kubernetes Security Context to automatically change the ownership of the volume. However, this feature does not work in all Kubernetes distributions.
 As an alternative, this chart supports using an initContainer to change the ownership of the volume before mounting it in the final destination.
 
-You can enable this initContainer by setting `volumePermissions.enabled` to `true`.
+You can enable this initContainer by setting `volumePermissions.enabled` to `true`. For OpenShift, one may either define the runAsUser and fsGroup accordingly, or try this more dynamic option: volumePermissions.securityContext.runAsUser="auto",securityContext.enabled=false,shmVolume.chmod.enabled=false.
```
Comment (juan131) [inline bitnami/influxdb/README.md:390]:
```suggestion
You can enable this **initContainer** by setting `volumePermissions.enabled` to `true`.
There are K8s distribution, such as OpenShift, where you can dynamically define the UID to run this **initContainer**. To do so, set the `volumePermissions.securityContext.runAsUser` to `auto`.
```

File: bitnami/influxdb/Chart.yaml (hunk 1 above)
Comment (mkerins) [reply bitnami/influxdb/Chart.yaml]: Fixed in latest commit

File: bitnami/influxdb/templates/influxdb/statefulset-high-availability.yaml (hunk 3 above)
Comment (mkerins) [reply bitnami/influxdb/templates/influxdb/statefulset-high-availability.yaml]: Fixed in latest commit

File: bitnami/influxdb/templates/influxdb/statefulset-high-availability.yaml (hunk 2 above)
Comment (mkerins) [reply bitnami/influxdb/templates/influxdb/statefulset-high-availability.yaml]: Fixed in latest commit

File: bitnami/influxdb/templates/influxdb/statefulset-high-availability.yaml (hunk 4 above)
Comment (mkerins) [reply bitnami/influxdb/templates/influxdb/statefulset-high-availability.yaml]: Removed the `xargs`, not necessary

File: bitnami/influxdb/README.md (hunk 5 above)
Comment (mkerins) [reply bitnami/influxdb/README.md]: Fixed in latest commit

**Comment by mkerins** [comment]
Thank you @juan131 , appreciate the feedback.

**Comment by rafariossaa** [comment]
Hi 
  Sorry @mkerins, during the weekend there was a upgrade of the chart and you need to solve the conflict in Chart.yaml.

**Comment by mkerins** [comment]
> Hi
> Sorry @mkerins, during the weekend there was a upgrade of the chart and you need to solve the conflict in Chart.yaml.

Updated

File: bitnami/influxdb/Chart.yaml [hunk 6]
```diff
@@ -15,4 +15,4 @@ maintainers:
 name: influxdb
 sources:
   - https://github.com/bitnami/bitnami-docker-influxdb
-version: 0.4.8
+version: 0.4.9
```
Comment (juan131) [inline bitnami/influxdb/Chart.yaml:18]: This should bump the minior version (`0.5.0`) but you're bumping the patch one

File: bitnami/influxdb/templates/influxdb/deployment-standalone.yaml [hunk 7]
```diff
@@ -64,16 +64,28 @@ spec:
           imagePullPolicy: {{ .Values.volumePermissions.image.pullPolicy | quote }}
           command:
             - /bin/bash
-            - -c
+            - -cx
```
Comment (juan131) [inline bitnami/influxdb/templates/influxdb/deployment-standalone.yaml:67]: Why do you add `-x`? It's for tracing, right ? See https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html
I wouldn't add it by default

Comment (mkerins) [reply bitnami/influxdb/templates/influxdb/deployment-standalone.yaml]: I had actually removed that based on feedback but I made a mistake when rebasing, sorry.

File: bitnami/influxdb/Chart.yaml (hunk 6 above)
Comment (mkerins) [reply bitnami/influxdb/Chart.yaml]: I'll get versions someday...

**Pull request merged**